Bug!!!  syntax parser in version 4.1.1

### environment :   sharding-jdbc-spring-boot-starter  4.1.1
### sql: update order_info set order_info.status=? where order_info.orderid=?


Exception in thread "main" org.apache.shardingsphere.underlying.common.exception.ShardingSphereException: Cannot support Multiple-Table for 'org.apache.shardingsphere.sql.parser.sql.statement.dml.UpdateStatement@6c56fff'.
	at org.apache.shardingsphere.sharding.route.engine.type.standard.ShardingStandardRoutingEngine.route(ShardingStandardRoutingEngine.java:72)
	at org.apache.shardingsphere.sharding.route.engine.ShardingRouteDecorator.decorate(ShardingRouteDecorator.java:69)
	at org.apache.shardingsphere.sharding.route.engine.ShardingRouteDecorator.decorate(ShardingRouteDecorator.java:53)
	at org.apache.shardingsphere.underlying.route.DataNodeRouter.executeRoute(DataNodeRouter.java:91)
	at org.apache.shardingsphere.underlying.route.DataNodeRouter.route(DataNodeRouter.java:76)
	at org.apache.shardingsphere.underlying.pluggble.prepare.PreparedQueryPrepareEngine.route(PreparedQueryPrepareEngine.java:54)
	at org.apache.shardingsphere.underlying.pluggble.prepare.BasePrepareEngine.executeRoute(BasePrepareEngine.java:96)
	at org.apache.shardingsphere.underlying.pluggble.prepare.BasePrepareEngine.prepare(BasePrepareEngine.java:83)
	at org.apache.shardingsphere.shardingjdbc.jdbc.core.statement.ShardingPreparedStatement.prepare(ShardingPreparedStatement.java:183)
	at org.apache.shardingsphere.shardingjdbc.jdbc.core.statement.ShardingPreparedStatement.executeUpdate(ShardingPreparedStatement.java:131)

@307434169 `Cannot support Multiple-Table` verification logic has been optimized in version 5.0.0-alpha. You can follow what @RaigorJiang said, use `update order_info set status=? where orderid=?` to avoid this exception.

During the investigation, I found an abnormality in the rewriting of the table in the set clause, and I will fix it soon.
```
[INFO ] 14:17:54.500 [ShardingSphere-Command-6] ShardingSphere-SQL - Logic SQL: /* ApplicationName=DataGrip 2021.1 */ UPDATE t_order SET t_order.content = '111111' WHERE t_order.order_id = 1
[INFO ] 14:17:54.500 [ShardingSphere-Command-6] ShardingSphere-SQL - SQLStatement: MySQLUpdateStatement(orderBy=Optional.empty, limit=Optional.empty)
[INFO ] 14:17:54.500 [ShardingSphere-Command-6] ShardingSphere-SQL - Actual SQL: ds_1 ::: /* ApplicationName=DataGrip 2021.1 */ UPDATE t_order_1 SET t_order.content = '111111' WHERE t_order_1.order_id = 1
[ERROR] 14:17:54.502 [ShardingSphere-Command-6] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Unknown column 't_order.content' in 'field list'
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
```

--- a/shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-rewrite/src/test/resources/sharding/update.xml
+++ b/shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-rewrite/src/test/resources/sharding/update.xml
@@ -21,10 +21,20 @@
         <input sql="UPDATE t_account SET status = ? WHERE account_id = ?" parameters="'OK', 100" />
         <output sql="UPDATE t_account_0 SET status = ? WHERE account_id = ?" parameters="'OK', 100" />
     </rewrite-assertion>
+
+    <rewrite-assertion id="update_with_sharding_value_with_table_for_parameters">
+        <input sql="UPDATE t_account SET t_account.status = ? WHERE t_account.account_id = ?" parameters="'OK', 100" />
+        <output sql="UPDATE t_account_0 SET t_account_0.status = ? WHERE t_account_0.account_id = ?" parameters="'OK', 100" />
+    </rewrite-assertion>
     
     <rewrite-assertion id="update_with_sharding_value_for_literals">
         <input sql="UPDATE t_account SET status = 'OK' WHERE account_id = 100" />
         <output sql="UPDATE t_account_0 SET status = 'OK' WHERE account_id = 100" />
+    </rewrite-assertion>
+
+    <rewrite-assertion id="update_with_sharding_value_with_table_for_literals">
+        <input sql="UPDATE t_account SET t_account.status = 'OK' WHERE t_account.account_id = 100" />
+        <output sql="UPDATE t_account_0 SET t_account_0.status = 'OK' WHERE t_account_0.account_id = 100" />
     </rewrite-assertion>
     
     <rewrite-assertion id="update_without_sharding_value_for_parameters">
@@ -32,10 +42,22 @@
         <output sql="UPDATE t_account_0 SET status = ? WHERE amount = ?" parameters="'OK', 1000" />
         <output sql="UPDATE t_account_1 SET status = ? WHERE amount = ?" parameters="'OK', 1000" />
     </rewrite-assertion>
+
+    <rewrite-assertion id="update_without_sharding_value_with_table_for_parameters">
+        <input sql="UPDATE t_account SET t_account.status = ? WHERE t_account.amount = ?" parameters="'OK', 1000" />
+        <output sql="UPDATE t_account_0 SET t_account_0.status = ? WHERE t_account_0.amount = ?" parameters="'OK', 1000" />
+        <output sql="UPDATE t_account_1 SET t_account_1.status = ? WHERE t_account_1.amount = ?" parameters="'OK', 1000" />
+    </rewrite-assertion>
     
     <rewrite-assertion id="update_without_sharding_value_for_literals">
-        <input sql="UPDATE t_account SET status = 'OK' WHERE amount = 10" parameters="'OK', 1000" />
-        <output sql="UPDATE t_account_0 SET status = 'OK' WHERE amount = 10" parameters="'OK', 1000" />
-        <output sql="UPDATE t_account_1 SET status = 'OK' WHERE amount = 10" parameters="'OK', 1000" />
+        <input sql="UPDATE t_account SET status = 'OK' WHERE amount = 10" />
+        <output sql="UPDATE t_account_0 SET status = 'OK' WHERE amount = 10" />
+        <output sql="UPDATE t_account_1 SET status = 'OK' WHERE amount = 10" />
+    </rewrite-assertion>
+
+    <rewrite-assertion id="update_without_sharding_value_with_table_for_literals">
+        <input sql="UPDATE t_account SET t_account.status = 'OK' WHERE t_account.amount = 10" />
+        <output sql="UPDATE t_account_0 SET t_account_0.status = 'OK' WHERE t_account_0.amount = 10" />
+        <output sql="UPDATE t_account_1 SET t_account_1.status = 'OK' WHERE t_account_1.amount = 10" />
     </rewrite-assertion>
 </rewrite-assertions>

--- a/shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/dml/UpdateStatementContextTest.java
+++ b/shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/dml/UpdateStatementContextTest.java
@@ -17,6 +17,7 @@
 
 package org.apache.shardingsphere.infra.binder.statement.dml;
 
+import org.apache.shardingsphere.sql.parser.sql.common.segment.dml.assignment.SetAssignmentSegment;
 import org.apache.shardingsphere.sql.parser.sql.common.segment.dml.column.ColumnSegment;
 import org.apache.shardingsphere.sql.parser.sql.common.segment.dml.expr.BinaryOperationExpression;
 import org.apache.shardingsphere.sql.parser.sql.common.segment.dml.predicate.WhereSegment;
@@ -32,6 +33,7 @@ import org.mockito.Mock;
 import org.mockito.junit.MockitoJUnitRunner;
 
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Optional;
@@ -66,6 +68,7 @@ public final class UpdateStatementContextTest {
         UpdateStatement updateStatement = new MySQLUpdateStatement();
         updateStatement.setWhere(whereSegment);
         updateStatement.setTableSegment(joinTableSegment);
+        updateStatement.setSetAssignment(new SetAssignmentSegment(0, 0, Collections.emptyList()));
         UpdateStatementContext actual = new UpdateStatementContext(updateStatement);
         assertThat(actual.getTablesContext().getTables().stream().map(a -> a.getTableName().getIdentifier().getValue()).collect(Collectors.toList()), is(Arrays.asList("tbl_1", "tbl_2")));
         assertThat(actual.getWhere(), is(Optional.of(whereSegment)));

--- a/shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/extractor/TableExtractor.java
+++ b/shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/extractor/TableExtractor.java
@@ -233,6 +233,7 @@ public final class TableExtractor {
      */
     public void extractTablesFromUpdate(final UpdateStatement updateStatement) {
         extractTablesFromTableSegment(updateStatement.getTableSegment());
+        updateStatement.getSetAssignment().getAssignments().forEach(each -> extractTablesFromExpression(each.getColumn()));
         if (updateStatement.getWhere().isPresent()) {
             extractTablesFromExpression(updateStatement.getWhere().get().getExpr());
         }

